src/Symfony/Component/Form/Extension/Core/Type/ChoiceType.php

```diff
@@ -158,6 +158,18 @@ public function buildForm(FormBuilderInterface $builder, array $options)
      */
     public function buildView(FormView $view, FormInterface $form, array $options)
     {
+        $choiceTranslationDomain = $options['choice_translation_domain'];
+        $translationDomain = $options['translation_domain'];
+        if (null === $choiceTranslationDomain) {
+            $choiceTranslationDomain = $translationDomain;
```

Comment:
You can remove this code by setting `choice_translation_domain` as a lazy option that returns the value of `$options['translation_domain']` by default. This way, this code is not executed when domain is overwritten.